### PR TITLE
add instructions for drf-spectacular to public api docs

### PR DESCRIPTION
**Pull request by JoshFerge**
- adds instructions for using the drf-spectacular library for api docs, added here https://github.com/getsentry/sentry/pull/30040

### CODE REVIEW

File: src/docs/api/public.mdx [hunk 1]
```diff
@@ -67,36 +67,70 @@ If your answers are Yes and No, you're in business - make the endpoint public. H
...
 You can find the tests here: [https://github.com/getsentry/sentry/tree/master/tests/apidocs](https://github.com/getsentry/sentry/tree/master/tests/apidocs).
 
 ### Local development
+
 **To add documentation for a new endpoint**:
+API Documentation consists of a few items:
+  - URL and query parameters
+  - Request Schema (None if no body)
+  - Response Schema
+  - Examples.
```
Comment (markstory) [inline src/docs/api/public.mdx:80]:
```suggestion
API Documentation consists of a few items:

- URL and query parameters
- Request Schema (None if no body)
- Response Schema
- Examples.
```

I don't know if the previous formatting would render correctly.

File: src/docs/api/public.mdx [hunk 2]
```diff
@@ -67,36 +67,70 @@ If your answers are Yes and No, you're in business - make the endpoint public. H
...
+```
+from sentry.apidocs.decorators import declare_public
+@declare_public(methods={"GET"})
+class OrganizationSCIMMemberDetails(...):
+```
+2. Tag the endpoint view if needed (you can also tag parent classes), this is how the endpoint makes it into a particular sidebar grouping.
+```
+from drf_spectacular.utils import extend_schema
+@extend_schema(tags=["SCIM"])
+class SCIMEndpoint(OrganizationEndpoint)...
```
Comment (markstory) [inline src/docs/api/public.mdx:94]:
```suggestion
from drf_spectacular.utils import extend_schema

@extend_schema(tags=["SCIM"])
class SCIMEndpoint(OrganizationEndpoint)...
```

File: src/docs/api/public.mdx [hunk 3]
```diff
@@ -67,36 +67,70 @@ If your answers are Yes and No, you're in business - make the endpoint public. H
...
+  - Response Schema
+  - Examples.
+ 
+Below we'll describe how to document these for an endpoint. You can familiarize yourself with some examples on the API docs website. We'll be making heavy use of `extend_schema` in order to do so.
+
+1. Declare the endpoint public using the `@declare_public` decorator, specifying the methods on the view which should be public.  
+```
+from sentry.apidocs.decorators import declare_public
+@declare_public(methods={"GET"})
+class OrganizationSCIMMemberDetails(...):
```
Comment (markstory) [inline src/docs/api/public.mdx:88]:
```suggestion
from sentry.apidocs.decorators import declare_public

@declare_public(methods={"GET"})
class OrganizationSCIMMemberDetails(...):
```

File: src/docs/api/public.mdx [hunk 4]
```diff
@@ -67,36 +67,70 @@ If your answers are Yes and No, you're in business - make the endpoint public. H
...
+   - Note that you can instead provide OpenAPI JSON, so you can fall back to this if you are running into issues.
+7. See [here](https://github.com/getsentry/sentry/blob/master/src/sentry/scim/endpoints/members.py) for an example of an endpoint documented using drf-spectacular, and the [drf-spectacular docs here](https://drf-spectacular.readthedocs.io/en/latest/) to learn more.
 
-1. Create a new file for your schema under the appropriate folder [here](https://github.com/getsentry/sentry/tree/master/api-docs/paths). Document the schema for the relevant requests.
-2. Add the new path to [this file](https://github.com/getsentry/sentry/blob/master/api-docs/openapi.json).
-3. Write a test [here](https://github.com/getsentry/sentry/tree/master/tests/apidocs).
+**Note that if the endpoint you're modifying previously had JSON documentation, you will need to delete the old documentation found in [this file](https://github.com/getsentry/sentry/blob/master/api-docs/openapi.json)**
 
+8. Write a test [here](https://github.com/getsentry/sentry/tree/master/tests/apidocs).
```
Comment (markstory) [inline src/docs/api/public.mdx:112]: With sublists and code blocks I would approach this as a set of paragraphs instead of a numbered list.

**Review by RyanSkonnord — APPROVED** [review]
LGTM

**Pull request merged**